es7/2017-01/jan-24.md

```diff
@@ -0,0 +1,1448 @@
+
+WH: ?.( is not a syntax issue. I don't think you should be calling functions on things that might not call a function.
+
+AWB: Isn't that your classic, "I want to call this method if it is there?"
+
+JL: I agree that the .? or whatever syntax we use, the property accessors we use are the most valuable. The rest of them seem like they have reasonable alternatives. 
+
+DH: To my mind, when you're adding syntax to the language it's for the most common idioms that you're trying to bless with the most concise notation. That you might ever want to say something is not the bar, it's that it's very common and idiomatic. I could be convinced otherwise, but pulling out a function from something that might not be there is much much less common than accessing a piece of data. In the cases where you might need to access a function you can handle using the existing && syntax but accessing field data is so common that we want concise syntax.
+
+JL: My motivation for the call was basically lots of setter/getter chains (getFoo()/setFoo(), not actually accessors). Maybe i should put together an example.
+
+MM: Concerned about semantics like `a?.b.c().d`, where the first "?" changes the meaning of all subsequent operators such as the  () and . attached to c and d. In this example c and d don't have ? operators and it's surprising to find out that, for example, c might not be called.
+
+DE: A nice thing about the surface syntax is that if you are accessing a private field on something other than this, you'll likely want to be able to do obj?.#x.
+
+AWB: I'm not sure we need to support this for private state.
+
+DE: If we went in the maximally general syntax, we want to consider
+
+DH: I think the chances are high this will be impossible
+
+MM: I'm trying to focus attention on a?.b.c().d does there need to be a?.b?.c().d? Should a single question mark color all further dots?
+
+BB: I agree, if you look at Swift, they support this and they require a ? after every optional. Docs: https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/OptionalChaining.html
+
+JHD: Now that we've considered this stage 1, I think we should table this until stage 2.
+
+GI: I will champion this in all forms
+
+AWB: Different threads went through different alternatives
+
+WH: I like it, but I don't want short-circuiting and free grouping.
+
+#### Conclusion/Resolution
+
+- Stage 1, champion  for investigating the whole space, not tied down to the details of Claude's concrete proposal.
+- Areas to investigate for next time:
+    - How should null be treated?
+    - Should we have a better default operator than || ?
+    - How should computed property access, function calls, new be handled--as in Claude's proposal, or another way? Are they needed for syntax--do they pay for themselves?
+    - Short-circuiting--does it need to be as general as in Claude's proposal. Also, free grouping.
+    - Interaction with private state--included or not?
```

Comment:
Maybe it'll be useful to mention an issue https://github.com/tc39/proposal-private-fields/issues/78